src/components/Chatbot/Chatbot.tsx

```diff
@@ -0,0 +1,334 @@
+/* eslint-disable no-nested-ternary */
+
+'use client';
+
+import { Button, Card, CardBody, CardFooter, CardHeader, Divider, Tooltip } from '@nextui-org/react';
+import clsx from 'clsx';
+import { Bot, Maximize2, Minimize2, PanelLeftClose, PanelLeftOpen, Send, X } from 'lucide-react';
+import { useEffect, useRef, useState } from 'react';
+
+import TypingText from '@/components/TypingText/TypingText';
+import {
+  CHAT_TITLE,
+  DATA_SOURCES,
+  DEFAULT_DATA_SOURCES,
+  DEFAULT_PROMPT,
+  ENTER_FULL_SCREEN,
+  EXIT_FULL_SCREEN,
+  SUB_WELCOME_MESSAGE,
+  TYPING_PLACEHOLDER,
+  WELCOME_MESSAGE,
+} from '@/domain/constant/chatbot/Chatbot';
+import { IChat } from '@/domain/entities/chatbot/Chatbot';
+import { SenderRole } from '@/domain/enums/SenderRole';
+import { APIError, chatService, formatChatResponse } from '@/services/api/chatbot';
+import { useMediaQuery } from '@/utils/resolution';
+
+import TypingDots from '../TypingText/TypingDot';
+import ChatbotSidebar from './ChatbotSidebar';
+
+export default function HungerMapChatbot() {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isFullScreen, setIsFullScreen] = useState(false);
+  const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+  const [chats, setChats] = useState<IChat[]>([{ id: 1, title: 'Chat 1', messages: [] }]);
+  const [currentChatIndex, setCurrentChatIndex] = useState(0);
+  const [input, setInput] = useState('');
+  const [isTyping, setIsTyping] = useState(false);
+  const inputRef = useRef<HTMLTextAreaElement>(null);
+  const chatEndRef = useRef<HTMLDivElement>(null);
+  // TODO: get isMobile from context later?
+  const isMobile = useMediaQuery('(max-width: 640px)');
```

Comment:
Currently no idea if we will get isMobile state from context or from nextjs? It is needed in js logic.